### DISCUSSION TITLE
Not writing to InfluxDB v2.4

### DISCUSSION THREAD

**Discussion by cliver1956** [johanmeijer/grott#193] [Q&A] [answered]
Hi,
I have grott running successfuly in a docker cotainer on my Asustor NAS alongside a seperate container running InfluxDB.
I have redirected the output of my Growatt SPH 6000TL BL-UP, or specifically my Shine WiFi-X Data Logger to point to it as described in your wiki.
It appears to be forwarding the data OK as I am still getting updated data from server.growatt.com.
However I don't seem to be seeing any data in my InfluxDB Bucket.
Using the Data Explorer in the InfluxDB web ui I can see the bucket I have set up ready to take the grott data but there is nothing to filter on.
Do I need to do anything in InfluxDB apart from create the bucket and copy the relevant Bucket Name, Token and Organisation ID to gather the data from grott?
One thing that springs to mind that I wasn't clear on is the format of the Organisation ID in the grott environmental variable. I have used the hex Organisation ID. Should it be the alpha Name instead?
The Docker run command I used that got grott running was:
`docker run -d -p 5279:5279 --restart on-failure -e gnomqtt="True" -e gblockcmd="False" -e gpvoutput="False" -e ginflux="True" -e ginflux2="True" -e gifip="xx.xxx.xxx.xxx" -e gifport="8086" -e giftoken="************ giforg="**********" -e gifbucket="****" --name grott ledidobe/grott`
I'm no Docker or InfluxDB expert so just trying to follow the documentation on both. 
So, any help appreciated.
I do have a bucket set up and taking data from an IotaWatt power monitor successfully. My plan is to use grott to give me my solar generation and battery charge/discharge stats alongside what I am getting from IotaWatt.

**Comment by johanmeijer** [comment]
Can you send a part of the log from the grott startup (if there are any messages related to influxdb) and at the moment grott tries to send something to influx? 

I do not use a hex organisation ID but my own defined in InfluxDB during setup.

**Comment by rany2** [comment]
The first thing I could think about is that this is a time zone issue. Grott isn't aware of the time zone that the datalogger is using.

Try setting TZ environment variable to the datalogger's time zone (in IANA format, so for example for Somali time set TZ to `Africa/Mogadishu`). So it would be:

` docker run -d -p 5279:5279 --restart on-failure -e TZ=Africa/Mogadishu -e gnomqtt="True" -e gblockcmd="False" -e gpvoutput="False" -e ginflux="True" -e ginflux2="True" -e gifip="xx.xxx.xxx.xxx" -e gifport="8086" -e giftoken="************ giforg="**********" -e gifbucket="****" --name grott ledidobe/grott`

**Comment by cliver1956** [comment]
Many thanks guys, I am getting a step closer.
I think I now have an InfluxDB setup problem because I now have the following in my container log:

```
	 - influxDB organization reynolds not defined or no authorisation to check

Grott settings:
```
I only have one organisation set up in my InfluxDB instance which is in use already running my IotaWatt bucket successfully.
Do I need a different organisation for each bucket? Or how to I give the current organisation the relevant permissions to access the grott bucket?

Or do I need to ask this on the InfluxDB forum?

**Reply by rany2** [reply]
Every organization has its own buckets, members, etc. Check which organization you created the bucket for grott in

**Reply by cliver1956** [reply]
When I created the bucket (using the webui) the only perameters I could enter where 'bucket name' and retention policy.
So, I deleted the bucket and recreated it using the CLI where I could set the org. 
The command I used was
`influx bucket create --name grottdb --org reynolds --retention 0 --token "my admin token"`
I then used the webui to create a grott-token which I used in the docker run file to re-create my container.
Again it runs but I get the same error
`- influxDB organization reynolds not defined or no authorisation to check`
I surely don't need to use the admin "all access" token do I? 
But, until I create the new bucket I can't create a rw token for it can I?
I know I must be missing something but I'm no influx expert, or docker for that matter. 
What is confusing me is that I am successfully using the same organisation to read and write data to an "iotawatt" bucket. The only difference is that the iotawatt setup uses the organisation ID which is a hex string created from the organisation name I entered when I set up influxdb to start with.

**Comment by cliver1956** [comment, marked as answer]
So, I decided to try deleting my influxdb container and start again setting the grott bucket up as my initial setup so I was guaranteed it would join the organisation. It worked. I am now loading data into my influxdb grottdb bucket.
Thanks to all for the help and sugestions.
Now to try the iotawatt setup.
